Tests/SwiftParserTest/Attributes.swift

```diff
@@ -36,21 +39,23 @@ final class AttributeTests: XCTestCase {
   func testMissingClosingParenToAttribute() {
     AssertParse(
       """
-      @_specialize(e#^DIAG_1^#
+      @_specialize(e#^DIAG^#
       """,
       diagnostics: [
-        DiagnosticSpec(locationMarker: "DIAG_1", message: "Expected ':' in attribute argument"),
-        DiagnosticSpec(locationMarker: "DIAG_1", message: "Expected ')' to end attribute"),
+        DiagnosticSpec(message: "Expected declaration"),
+        DiagnosticSpec(message: "Expected ':' in attribute argument"),
+        DiagnosticSpec(message: "Expected ')' to end attribute"),
```

Comment:
> What do you think about special-casing trailing missing nodes? Ie. If a parent has multiple sequential missing nodes, instead use `Expected complete <parent>`. Open to ideas for "complete" there. Ideally pretty printing (which we don't have yet) could then output an example/grammar/something else as well 🤔.

Let’s do this in a follow-up PR because it might be non-trivial

> I'd also expect an `after attribute` for `Expected declaration`, ie. `Expected declaration after attribute` +

Done

> for it to be after the attribute diagnostics.

This is non-trivial because we can’t just order the diagnostics by their `indexInTree` anymore. Let’s consider do it in a follow-up PR.